packages/workspace/src/merger/index.ts

```
@@ -97,3 +97,21 @@ export const mergeElements = async (
 
   return { merged, errors }
 }
+
+export const mergeSingleElement = async <T extends Element>(elementParts: T[]): Promise<T> => {
+  const mergeRes = await mergeElements(awu(elementParts))
+
+  const [error] = await awu(await mergeRes.errors.entries()).toArray()
```

again, why only show the first error and not all errors?
it is also a bit strange that the interface here throws an exception instead of returning errors like we tend to do.
This will have to be changed anyway in your next task I assume if you ever want to show the errors, but even if it didn't, this is not a good interface for infrastructure functions, we should try to be consistent with out API

---

I get why we would return an error instead of throwing it when the function can partially succeed or fail (e.g. `mergeElements` when it is possible to succeed with some of the elements and fail with others), but in this case, I don't think it will make much sense.

I think that even after the next task in some cases we will want it to be thrown to the user (for example, if the default configuration contains merge errors, or if the configuration suggestions from fetch contain merge errors)
Not throwing it means I will need to return the error from every usage of `mergeSingleElement` all the way to the CLI and changing the signature of every function along the way

Also, I see a lot of places in our API that throw errors so I am not sure it is inconsistent with the rest of our code

---

1. there are multiple errors here, if you throw, you can only throw one, that makes you lose information, especially in cases like we are going to have where we'd want to show these merge errors to the user, we would want to be consistent with the error interface, so we'd want these to be the same errors as mergeElements returns
2. we should not be throwing exceptions in cases that can reasonably happen and that our code should be able to handle (like merge errors in user input like elements of config), it is true that there are places in the API that currently throw exceptions as part of the API, those are bugs and should be fixed (for example - https://salto-io.atlassian.net/browse/SAAS-2641)
3. there can be exceptions in the code, in cases where there is really nothing else to do. either the expected behavior is to crash the CLI or in code that should never happen. that is different from an API where part of its legitimate result is an error or a list of errors